### PR TITLE
substrate_metadata documentation

### PR DESCRIPTION
**Pull request by justkawal**
## Description
* Adds readme.md file for `substrate_metadata`.
Preview: https://github.com/rankanizer/polkadart/tree/justkawal/document-substrate-metadata/packages/substrate_metadata

## Why?
To resolve issue: #135

### CODE REVIEW

**Comment by gabrielokura** [comment]
Should we have an example folder in this package too?

**Comment by justkawal** [comment]
> Should we have an example folder in this package too?

Yes, we should

**Comment by gabrielokura** [comment]
> 
 #144

**Comment by gabrielokura** [comment]
> > Should we have an example folder in this package too?
> 
> Yes, we should

#144

**Review by gabrielokura — APPROVED** [review]

File: packages/substrate_metadata/README.md [hunk 1]
```diff
@@ -1,3 +1,174 @@
...
+```css
+flutter pub get
+```
+
+### 3. Import it
+
+Now in your `Dart` code, you can use:
+
+```dart
+import 'package:polkadart_scale_codec/polkadart_scale_codec.dart';
```
Comment (gabrielokura) [inline packages/substrate_metadata/README.md:37]: Is this right? Shouldn't be 'substrate_metadata...'?

**Review by gabrielokura — DISMISSED** [review]

File: packages/substrate_metadata/README.md [hunk 2]
```diff
@@ -1,3 +1,174 @@
...
+```
+
+# Usage
+
+### Decode Metadata
+
+```dart
+  // create MetadataDecoder signleton instance
```
Comment (pastre) [inline packages/substrate_metadata/README.md:45]: typo in singleton 
```suggestion
  // create MetadataDecoder singleton instance
```

Comment (pastre) [inline packages/substrate_metadata/README.md:45]: Also you can't create a singleton's instance, only get its reference
```suggestion
  // get MetadataDecoder singleton instance
```

File: packages/substrate_metadata/README.md [hunk 3]
```diff
@@ -1,3 +1,174 @@
-## substrate_metadata
+# substrate_metadata
 
-Something cool is happening...
+[substrate_metadata](https://www.pub.dev/packages/substrate_metadata) is a flutter and dart library for encoding and decoding chain **metadata**, **constants**, **extrinsic** and **events** of blocks.
```
Comment (leonardocustodio) [inline packages/substrate_metadata/README.md:3]: extrinsics in the plural

File: packages/substrate_metadata/README.md [hunk 4]
```diff
@@ -1,3 +1,174 @@
...
+
+# Lets Get Started
+
+### 1. Depend on it
+
+Add this to your package's `pubspec.yaml` file:
+
+```yaml
+dependencies:
+  substrate_metadata: 0.0.2
```
Comment (leonardocustodio) [inline packages/substrate_metadata/README.md:13]: not sure we should use a fixed version here we will have to remember everytime to update here in that case

**Review by leonardocustodio — CHANGES_REQUESTED** [review]

File: packages/substrate_metadata/README.md (hunk 4 above)
Comment (pastre) [reply packages/substrate_metadata/README.md:13]: I think having it fixed is safer. We could have a CI step that updates that line and checks for a changelog

Comment (justkawal) [reply packages/substrate_metadata/README.md:13]: I think `any` would do the work.

File: packages/substrate_metadata/README.md (hunk 2 above)
Comment (justkawal) [reply packages/substrate_metadata/README.md:45]: Done

Comment (justkawal) [reply packages/substrate_metadata/README.md:45]: Done

File: packages/substrate_metadata/README.md (hunk 1 above)
Comment (justkawal) [reply packages/substrate_metadata/README.md:37]: Done

**Review by leonardocustodio — APPROVED** [review]

**Review by pastre — APPROVED** [review]

**Pull request merged**